### PR TITLE
Add show_dtype support for plot_model, update related tests. 

### PR DESCRIPTION
**Pull request by jonah-kohn**
In response to a feature request.

### CODE REVIEW

File: tensorflow/python/keras/utils/vis_utils.py [hunk 1]
```diff
@@ -180,6 +188,16 @@ def model_to_dot(model,
       label = '{}: {}'.format(layer_name, class_name)
     else:
       label = class_name
+  
+    # Rebuild the label as a table including the layer's dtype.
+    if show_dtype: 
+      def format_dtype(dtype):
```
Comment (qlzh727) [inline tensorflow/python/keras/utils/vis_utils.py:194]: It seems that this function is not used here, should the label below be:

```
label = '%s|%s' % (label, format_dtype(dtype))
```

**Review by qlzh727 — CHANGES_REQUESTED** [review]
Thanks Jonah for the contribution.

File: tensorflow/python/keras/utils/vis_utils.py (hunk 1 above)
Comment (jonah-kohn) [reply tensorflow/python/keras/utils/vis_utils.py:194]: Definitely, Scott. Thank you!

File: tensorflow/python/keras/utils/vis_utils.py [hunk 2]
```diff
@@ -180,6 +188,16 @@ def model_to_dot(model,
...
+          if dtype is None:
+            return '?'
+          else:
+            return str(dtype)
+          
+      label = '%s|%s' % (label, dtype)
```
Comment (jonah-kohn) [inline tensorflow/python/keras/utils/vis_utils.py:200]:
```suggestion
      label = '%s|%s' % (label, format_dtype(dtype))
```

**Review by jonah-kohn — COMMENTED** [review]
Implemented Scott's suggested change

File: tensorflow/python/keras/utils/vis_utils.py [hunk 3]
```diff
@@ -180,6 +188,16 @@ def model_to_dot(model,
...
+  
+    # Rebuild the label as a table including the layer's dtype.
+    if show_dtype: 
+      def format_dtype(dtype):
+          if dtype is None:
+            return '?'
+          else:
+            return str(dtype)
+          
+      label = '%s|%s' % (label, format_dtype(dtype))
```
Comment (jonah-kohn) [inline tensorflow/python/keras/utils/vis_utils.py:200]:
```suggestion
      label = '%s|%s' % (label, format_dtype(layer.dtype))
```

**Review by jonah-kohn — COMMENTED** [review]
Correctly access layer attributes

**Review by qlzh727 — DISMISSED** [review]

File: tensorflow/python/keras/utils/vis_utils.py [hunk 4]
```diff
@@ -180,6 +188,16 @@ def model_to_dot(model,
...
     else:
       label = class_name
+  
+    # Rebuild the label as a table including the layer's dtype.
+    if show_dtype: 
+      def format_dtype(dtype):
+          if dtype is None:
+            return '?'
+          else:
+            return str(dtype)
```
Comment (jonah-kohn) [inline tensorflow/python/keras/utils/vis_utils.py:198]:
```suggestion
        if dtype is None:
          return '?'
        else:
          return str(dtype)
```

**Review by jonah-kohn — COMMENTED** [review]
Fit linter indentation requirements.

File: tensorflow/python/keras/optimizer_v2/optimizer_v2.py [hunk 5]
```diff
@@ -910,7 +910,7 @@ def _serialize_hyperparameter(self, hyperparameter_name):
       return value()
     if tensor_util.is_tensor(value):
       return backend.get_value(value)
-    return value
+    return float(value)
```
Comment (qlzh727) [inline tensorflow/python/keras/optimizer_v2/optimizer_v2.py:913]: Not sure why this is included in this PR.

Comment (jonah-kohn) [reply tensorflow/python/keras/optimizer_v2/optimizer_v2.py:913]:
```suggestion
    return value
```
It shouldn't be, pushed to the wrong branch. Sorry for the unnecessary delay.

**Review by qlzh727 — DISMISSED** [review]

File: tensorflow/python/keras/utils/vis_utils_test.py [hunk 6]
```diff
@@ -88,7 +91,8 @@ def test_plot_model_with_add_loss(self):
     dot_img_file = 'model_4.png'
     try:
       vis_utils.plot_model(
-          model, to_file=dot_img_file, show_shapes=True, expand_nested=True)
+          model, to_file=dot_img_file, show_shapes=True,
+          show_dtype=True expand_nested=True)
```
Comment (jonah-kohn) [inline tensorflow/python/keras/utils/vis_utils_test.py:95]:
```suggestion
          show_dtype=True, expand_nested=True)
```

Comment (jonah-kohn) [reply tensorflow/python/keras/utils/vis_utils_test.py:95]: Response to sanity check logs

**Review by qlzh727 — APPROVED** [review]
